src/ui/pages.rs

```diff
@@ -56,3 +76,16 @@ pub fn chat_page(parent_stack: Rc<Stack>, token_data: LoginInfo) {
 
     parent_stack.add_named(&sections, Some("chats"));
 }
+
+async fn get_login_by_token(url: &str, header:HeaderValue) -> Result<String, Box<dyn std::error::Error>> {
```

Comment:
This feels like a very generic function which will commonly be used to communicate to discord so you should probably create `src/discord/utils.rs` file and put it there